### PR TITLE
Akka 2.5.23 and Scala 2.13.0-RC2

### PR DESCRIPTION
**Pull request by renatocaval**

### CODE REVIEW

**Comment by renatocaval** [comment]
So, build is failing missing Scala 2.13 artefacts for Akka 2.5.19. 

Shall we bump the version here to Akka 2.6.0-M2 in order to get the artefacts? We do need 2.6.0-Mx anyway.

File: build.sbt [hunk 1]
```diff
@@ -32,20 +32,27 @@ def common: Seq[Setting[_]] = Seq(
...
-    "-Xfuture",
     "-Xfatal-warnings"
   ),
+  scalacOptions ++= {
+    // define scalac options that are only valid or desirable for 2.11 and 2.12
+    if (scalaVersion.value.startsWith("2.13")) Seq.empty
+    else 
+    Seq(
+      "-Xfuture", // invalid in 2.13
+      "-deprecation" // temporarily allowing deprecation because of Java colletion coverters
```
Comment (renatocaval) [inline build.sbt]: This is annoying. `scala.collection.JavaConverters` is deprecated in 2.13 and instead, we should use `scala.jdk.CollectionConverters`.

So for now, I'm disabling the `-deprecation` flag for 2.13.

**Comment by renatocaval** [comment]
I originally bumped the version to Akka 2.6.0-M2, but moved now to 2.5.23 because there is no 2.11 artefacts for 2.6.x.

File: build.sbt (hunk 1 above)
Comment (dwijnand) [reply build.sbt]: Opened https://github.com/scala/scala-collection-compat/issues/208 to track this discussion.  Talked about it with the Scala Team, I'll jot down my notes there.

**Comment by patriknw** [comment]
> I originally bumped the version to Akka 2.6.0-M2, but moved now to 2.5.23 because there is no 2.11 artefacts for 2.6.x.

and we want to support Akka 2.5.x applications without having to maintain two separate branches here, so 2.5.23 is correct

**Comment by renatocaval** [comment]
Is that normal? No 2.11 builds for Akka 2.5.23?

```
[warn] 	::          UNRESOLVED DEPENDENCIES         ::
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	:: com.typesafe.akka#akka-persistence-typed_2.11;2.5.23: not found
[warn] 	:: com.typesafe.akka#akka-cluster-typed_2.11;2.5.23: not found
[warn] 	:: com.typesafe.akka#akka-actor-testkit-typed_2.11;2.5.23: not found
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
```

**Comment by renatocaval** [comment]
Seems to be only the typed binaries that were not built for 2.11 in Akka 2.5.23.

Was that an oversight?

**Comment by dwijnand** [comment]
Huh, the last release of http://repo1.maven.org/maven2/com/typesafe/akka/akka-persistence-typed_2.11/ is 2.5.21.

**Comment by patriknw** [comment]
It's as intended. We dropped 2.11 for -typed modules because there was something that we needed that wasn't backported to 2.11 (for good reasons)

**Comment by dwijnand** [comment]
Hmm.  So 2.5.21 is the last release for 2.11 and 2.5.23 is the first release for 2.13.0-RC2.

I think this will be OK.

Play 2.7 supports Scala 2.11 and 2.12, and we'd like to cut a 2.7.x release for 2.13 too (soon 🤞) and we should be able to.

Lagom 1.5 supports Scala 2.11 and 2.12, while Lagom 1.6 drops support for 2.11.  I think even if the problem here were resolved we'd run into other blockers.  So I think Scala 2.13 in Lagom 1.6+ is fine.

So I think we should just drop the cross-build for 2.11 here.  New version series 0.7x?

**Comment by TimMoore** [comment]
The dependencies on the typed modules are all for the benefit of one test class. Another option could be to split that test into its own module and only build/run it with 2.12, or something like that.

**Comment by dwijnand** [comment]
Yeah, Patrik mentioned that it was test-only and mentioned maybe releasing for 2.11 without tests.  Your solution avoids not running most tests.

We decided that we're ok with dropping 2.11.  I can have a go at your idea after this PR is merged.

**Comment by ignasi35** [comment]
I can't push to this branch. I'll open a new PR on my fork that supersedes this one.

PS: can't close the PR either.

**Comment by 2m** [comment]
Supersede by #484

**Pull request closed without merging**